### PR TITLE
[bitnami/keycloak] Keycloak 17

### LINKED ISSUE #9061: Support Keycloak in Quarkus Mode
### Name and Version

bitnami/keycloak latest

### What is the problem this feature will solve?

Keycloak distribution based on WildFly is deprecated and has switched to Quarkus. The CLI and Docker images are very different, so the current chart doesn't work.

### What is the feature you are proposing to solve the problem?

Update the chart to use the Quarkus distribution, or introduce a separate version: keycloak-wildlfy / keycloak-quarkus.

### What alternatives have you considered?

_No response_

### PR DESCRIPTION
**Pull request by Mauraza**
### Description of the change

Updating the chart to version 17, now with Quarkus

### Applicable issues

  - fixes #9061

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/keycloak/templates/statefulset.yaml [hunk 1]
```diff
@@ -294,23 +294,23 @@ spec:
           {{- if .Values.startupProbe.enabled }}
           startupProbe: {{- omit .Values.startupProbe "enabled" | toYaml | nindent 12 }}
             httpGet:
-              path: /auth/
```
Comment (FraPazGal) [inline bitnami/keycloak/templates/statefulset.yaml:297]: Has this path been changed in the new major? We are using it in our [cypress tests](    cy.visit('/auth/admin');) and we should modify them if so.

File: bitnami/keycloak/values.yaml [hunk 2]
```diff
@@ -70,7 +70,7 @@ diagnosticMode:
 image:
   registry: docker.io
   repository: bitnami/keycloak
-  tag: 16.1.1-debian-10-r85
+  tag: 17.0.1-debian-10-r0
```
Comment (FraPazGal) [inline bitnami/keycloak/values.yaml:73]: Not an issue, but we can leave this as it is. The internal CI will update these tags regardless.

**Review by FraPazGal — CHANGES_REQUESTED** [review]
Thanks for the PR @Mauraza! I have some concerns with the breaking changes introduced in this new major, could you please look at my comments?

File: bitnami/keycloak/values.yaml [hunk 3]
```diff
@@ -227,16 +227,6 @@ serviceDiscovery:
   ## @param serviceDiscovery.transportStack Transport stack for the discovery protocol set in `serviceDiscovery.protocol` parameter
   ##
   transportStack: tcp
-## Keycloak cache settings
-## ref: https://github.com/bitnami/bitnami-docker-keycloak#cluster-configuration
-##
-cache:
-  ## @param cache.ownersCount Number of nodes that will replicate cached data
-  ##
-  ownersCount: 1
```
[the remark this answers is not in the document]
Comment (Mauraza) [reply bitnami/keycloak/values.yaml:236]: The cache was changed https://www.keycloak.org/server/caching

**Review by FraPazGal — DISMISSED** [review]
LGTM!

**Review by FraPazGal — DISMISSED** [review]

**Pull request merged**